ctapipe/core/traits.py

```diff
@@ -138,22 +139,51 @@ def has_traits(cls, ignore=("config", "parent")):
     return bool(set(cls.class_trait_names()) - set(ignore))
 
 
+class TelescopePatternList(UserList):
+    """
+    Representation for a list of telescope pattern tuples. This is a helper class
+    used  by the Trait TelescopeParameter as its value type
+    """
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self._lookup = None
+        self._old_getitem = self.__getitem__
+        self._subarray = None
```

Comment:
Why is the subarray stored here and in lookup?

Reply:
That was for allowing the subarray to be re-applied automatically on change, but that works without it now. Will remove.